crates/sui-core/src/authority_server.rs

```diff
@@ -453,22 +465,30 @@ impl ValidatorService {
 
         let shared_object_tx = certificates[0].contains_shared_object();
 
-        let _metrics_guard = if wait_for_effects {
-            if shared_object_tx {
-                self.metrics
-                    .handle_certificate_consensus_latency
-                    .start_timer()
+        let metrics = if certificates.len() == 1 {
+            if wait_for_effects {
+                if shared_object_tx {
+                    &self.metrics.handle_certificate_consensus_latency
+                } else {
+                    &self.metrics.handle_certificate_non_consensus_latency
+                }
             } else {
-                self.metrics
-                    .handle_certificate_non_consensus_latency
-                    .start_timer()
+                &self.metrics.submit_certificate_consensus_latency
             }
         } else {
-            self.metrics
-                .submit_certificate_consensus_latency
-                .start_timer()
+            if wait_for_effects {
+                &self
+                    .metrics
+                    .handle_soft_bundle_certificates_consensus_latency
+            } else {
+                &self
+                    .metrics
+                    .handle_soft_bundle_certificates_non_consensus_latency
+            }
```

Comment:
This branch should check if any transaction in the bundle contains a shared object. And  I don't think we need to add a metric corresponding to `submit_certificate_consensus_latency` (`!wait_for_effects`) for soft bundle for now.

Reply:
I have added a comment here saying that the check has already been ensured by `soft_bundle_validity_check`, so in practice all certificates will contain shared objects when entering this branch.  Since that is the case, also removed `handle_soft_bundle_certificates_non_consensus_latency`.